### PR TITLE
feat: rollup plugin support generate file with pattern

### LINKED ISSUE #778: [hash] in fileName
### Describe the feature request

Hello, I'm using the rollup plugin and I went to output the css file with a hash, I tried adding `[hash]` in `fileName` but it doesn't get replaced by a hash, can this feature be supported, if yes, how to do it (I can also implement it and make PR).

Many thanks.

```js
import stylexPlugin from '@stylexjs/rollup-plugin';

const config = {
  input: './index.js',
  output: {
    file: './.build/bundle.js',
    format: 'es',
  },
  // Ensure that the stylex plugin is used before Babel
  plugins: [stylexPlugin({
    // Required. File path for the generated CSS file.
    fileName: './.build/stylex.[hash].css',
    // default: false
    dev: false,
    // prefix for all generated classNames
    classNamePrefix: 'x',
    // Required for CSS variable support
    unstable_moduleResolution: {
      // type: 'commonJS' | 'haste'
      // default: 'commonJS'
      type: 'commonJS',
      // The absolute path to the root directory of your project
      rootDir: __dirname,
    },
  })],
};

export default config;
```

### PR DESCRIPTION
**Pull request by nonzzz**
## What changed / motivation ?

Until now I found that no one has implemented it.

## Linked PR/Issues

Fixes #778 

## Additional Context

Screenshots, Tests, Anything Else

## Pre-flight checklist

- [x] I have read the contributing guidelines
      [Contribution Guidelines](https://github.com/facebook/stylex/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

File: packages/rollup-plugin/src/index.js [hunk 1]
```diff
@@ -44,6 +45,15 @@ export type PluginOptions = $ReadOnly<{
...
 }>;
 
+function replaceFileName(original: string) {
+  const hash = crypto
+    .createHash('sha256')
+    .update(crypto.randomBytes(16))
+    .digest('hex')
+    .slice(0, 8);
+  return original.replace(/\[hash\]/g, hash);
+}
```
Comment (nmn) [inline packages/rollup-plugin/src/index.js:55]:
```suggestion
function replaceFileName(original: string) {
  if (!original.includes('[hash]')) {
    return original;
  }
  const hash = crypto
    .createHash('sha256')
    .update(crypto.randomBytes(16))
    .digest('hex')
    .slice(0, 8);
  return original.replace(/\[hash\]/g, hash);
}
```

Comment (nmn) [reply packages/rollup-plugin/src/index.js:55]: Also, shouldn't the hash be hashing the contents of the generated file?

**Review by nmn — CHANGES_REQUESTED** [review]
Some small changes suggested.

File: packages/rollup-plugin/src/index.js (hunk 1 above)
Comment (nonzzz) [reply packages/rollup-plugin/src/index.js:55]: No, in my opinion that feature request is on hashing the generated file name not the contents of the generated file. By the way following rollup document, we can use `assetFileNames` do this thing but it need to split chunk by themselves. (But we need change the emit logic)

Comment (nmn) [reply packages/rollup-plugin/src/index.js:55]: from everything I've read the point of putting a hash in the file name is for a cache breaker, which should likely be based on the file contents.

Comment (nonzzz) [reply packages/rollup-plugin/src/index.js]: My bad, now i update the code :)

**Comment by nmn** [comment]
Thanks!

**Pull request merged**